demux/demux_playlist.c

```diff
@@ -35,6 +36,31 @@
 
 #define PROBE_SIZE (8 * 1024)
 
+enum dir_mode {
+    DIR_RECURSIVE,
+    DIR_LAZY,
+    DIR_IGNORE,
+};
+
+#define OPT_BASE_STRUCT struct demux_playlist_opts
+struct demux_playlist_opts {
+    int dir_mode;
+};
+
+struct m_sub_options demux_playlist_conf = {
+    .opts = (const struct m_option[]) {
+        {"directory-mode", OPT_CHOICE(dir_mode,
+            {"recursive", DIR_RECURSIVE},
+            {"lazy", DIR_LAZY},
+            {"ignore", DIR_IGNORE})},
+        {0}
+    },
+    .size = sizeof(struct demux_playlist_opts),
+    .defaults = &(const struct demux_playlist_opts){
+        .dir_mode = 0
+    },
```

Comment:
```suggestion
    .defaults = &(const struct demux_playlist_opts) {0},
```

For consistency mainly.